src/Digdir.Domain.Dialogporten.Janitor/Features/UpdateSubjectResources/UpdateSubjectResources.cs

```diff
@@ -0,0 +1,132 @@
+        var lastUpdated = since ?? _db.SubjectResourceLastUpdates.OrderByDescending(x => x.LastUpdate)
+            .FirstOrDefault()?.LastUpdate ?? DateTimeOffset.MinValue;
+
+        // Allow for a few seconds of drift between server clocks
+        var newLastUpdated = DateTimeOffset.Now.Subtract(TimeSpan.FromSeconds(3));
+
+        _logger.LogInformation("Fetching updated subject resources since {LastUpdated}", lastUpdated);
+
+        // 2. Get the resources from the resource registry, supplying the last updated as "since"
+        var remoteUpdatedSubjectResources =
+            await _resourceRegistry.GetUpdatedSubjectResources(lastUpdated, cancellationToken);
+
+        _logger.LogInformation("Fetched {Count} updated subject resources", remoteUpdatedSubjectResources.Count);
+
+        // 3. Delete the subjectResources that are no longer in the resource registry
+        var remoteDeletedSubjectResources = remoteUpdatedSubjectResources
+            .Where(x => x.Deleted)
+            .Select(deletedSubjectResource => new
+            {
+                deletedSubjectResource.Resource,
+                deletedSubjectResource.Subject
+            })
+            .ToList();
+
+        if (remoteDeletedSubjectResources.Count != 0)
+        {
+            var resourceStrings = remoteDeletedSubjectResources.Select(r => r.Resource.ToString()).ToList();
+            var subjectStrings = remoteDeletedSubjectResources.Select(r => r.Subject.ToString()).ToList();
+
+            var localSubjectResourcesToDelete = _db.SubjectResources
+                .Where(x => resourceStrings.Contains(x.Resource) && subjectStrings.Contains(x.Subject))
+                .ToList();
+
+            if (localSubjectResourcesToDelete.Count != 0)
+            {
+                _logger.LogInformation("Deleting {Count} subject resources", localSubjectResourcesToDelete.Count);
+                _db.SubjectResources.RemoveRange(localSubjectResourcesToDelete);
+            }
+        }
+
+        // 4. Add the ones that are new (or no longer deleted)
+        var remoteNewSubjectResources = remoteUpdatedSubjectResources
+            .Where(x => !x.Deleted)
+            .Select(updatedSubjectResource => new SubjectResource
+            {
+                Id = Uuid7.NewUuid7().ToGuid(matchGuidEndianness: true),
+                Resource = updatedSubjectResource.Resource.ToString(),
+                Subject = updatedSubjectResource.Subject.ToString(),
+                // We use the updated timestamp from the remote API as our CreatedAt timestamp
+                // UpdatedAt at our end is the time we received the data
+                CreatedAt = updatedSubjectResource.UpdatedAt
+            })
+            .ToList();
+
+        if (remoteNewSubjectResources.Count != 0)
+        {
+            // FIXME! This is obviously horribly inefficient as the table grows
+            var localExistingResources = _db.SubjectResources
```

Comment:
Trenger bistand fra noen med bedre EF-fu skills enn meg til å gjøre dette mer effektivt. Her er det behov for å kunne gjøre noe upserts noe ala `INSERT ... ON CONFLICT UPDATE ...` for å kunne håndtere at DP har rader i sin kopi som RR anser som nye/oppdaterte.